lib/pulselistener.js

```diff
@@ -49,14 +59,42 @@ var retryConnect = function(connectionString, retries) {
 /**
  * Create PulseConnection from `options` on the form:
  * {
- *   namespace:         // Namespace to prefix queues/exchanges (optional)
- *                      // defaults to `username` if given otherwise ""
- *   username:          // Username to connect with (and namespace if not given)
- *   password:          // Password to connect with
- *   hostname:          // Hostname to connect to using username/password
- *                      // defaults to pulse.mozilla.org
- *   connectionString:  // connectionString cannot be used with username,
- *                      // password and/or hostname.
+ *   namespace:         // Namespace to prefix queues/exchanges (optional
+ *                      // when using connectionString or username/password,
+ *                      // defaults to `username` if given otherwise "";
+ *                      // required when using taskcluster-pulse)
+ *   // There are 3 exclusive ways to configure Pulse credentials. Please only
+ *   // use one of them - if properties belonging to different groups are
+ *   // provided at the same time, exceptions will be raised.
+ *   // 1. connectionString
+ *   connectionString:  // An AQMP URI containing the protocol, username,
+ *                      // password, hostname, etc. to be used directly.
+ *   // 2. Pulse username and password (deprecated)
+ *   username:          // Pulse username
+ *   password:          // Pulse password
+ *   hostname:          // Hostname to connect to (optional, defaults to
+ *                      // pulse.mozilla.org)
+ *   // 3. Taskcluster-Pulse (recommended; when only namespace is provided,
+ *   // this method will be used)
+ *   credentials:       // *Taskcluster* credentials (optional, defaults to
+ *                      // the TC credentials from env)
+ *   baseUrl:           // Base URL of taskcluster-pulse (optional, defaults
+ *                      // to https://pulse.taskcluster.net/v1)
+ *   // Contact info sent to taskcluster-pulse when requesting credentials
+ *   // (optional, defaults to process.env.OWNER_EMAIL)
+ *   contact: {         // One of:
```

Comment:
Good idea. It'd be more suitable to go into another PR later I think as it's kinda independent and compatible with the current implementation.